src/tls/openssl.c

```diff
@@ -363,6 +366,10 @@ static int tls_net_read(struct flb_upstream_conn *u_conn,
         else if (ret == SSL_ERROR_WANT_WRITE) {
             ret = FLB_TLS_WANT_WRITE;
         }
+        else if (ret < 0) {
+            ERR_error_string(ret, errbuf);
```

Comment:
How about using `ERR_error_string_n` ?
https://www.openssl.org/docs/man3.0/man3/ERR_error_string.html
```
Note that this function is not thread-safe and does no checks on the size of the buffer; use ERR_error_string_n() instead.
```

Reply:
I changed all uses of ERR_error_string to ERR_error_string_n.